cpplite/cpplite.debugger/src/org/netbeans/modules/cpplite/debugger/models/VariablesModel.java

```diff
@@ -96,24 +100,31 @@ public Object getRoot () {
      */
     @Override
     public Object[] getChildren (Object parent, int from, int to) throws UnknownTypeException {
-        CPPVariable parentVar;
+        NIVariable parentVar;
         if (parent == ROOT) {
             parentVar = null;
-        } else if (parent instanceof CPPVariable) {
-            parentVar = (CPPVariable) parent;
+        } else if (parent instanceof NIVariable) {
+            parentVar = (NIVariable) parent;
         } else {
             throw new UnknownTypeException (parent);
         }
         CPPFrame frame = currentFrame;
         if (frame != null) {
-            Map<String, CPPVariable> variables = (parentVar == null) ? frame.getVariables() : parentVar.getChildrenVariables();
-            Object[] array = variables.values().toArray();
-            if (array.length == 1 && array[0] == null) {
-                // Some error / message
-                return new Object[]{variables.keySet().iterator().next()};
+            NIVariable[] array;
+            if (parentVar == null) {
+                Map<String, NIVariable> variables = frame.getVariables();
+                array = variables.values().toArray(new NIVariable[0]);
+                if (array.length == 1 && array[0] == null) {
+                    // Some error / message
+                    return new Object[]{variables.keySet().iterator().next()};
+                }
+                if (displayer != null) {
+                    array = displayer.displayed(array);
+                }
             } else {
-                return array;
+                array = parentVar.getChildren(from, to);
```

Comment:
Shouldn't the displayer be applied to `parentVar` as well?

Reply:
Good point, done.